socorro/external/postgresql/setupdb_app.py

```diff
@@ -219,9 +232,48 @@ def generate_fakedata(self, db, fakedata_days):
             """, dict(zip(["one", "two", "three", "four"],
                       list(fakedata.featured_versions))))
 
-    def main(self):
+    def construct_db_url(self, dbname=None, superuser=False):
+        """Takes a URL to connect to Postgres and updates database name
+            or superuser name/password as indicated"""
+        if self.database_url:
+            database_url = self.database_url
+            url = urlparse.urlparse(database_url)
+            database_username = url.username
+            database_password = url.password
+            database_hostname = url.hostname
+            database_port = url.port
+            # Set the self.database_name because it wasn't set
+            # intentionally by --database_name if database_url
+            # is defined
+            self.database_name = url.path[1:]
```

Comment:
Hurts my soul that this is the third time I see `urlparse()` being used. Can we simply just add a `def parse_database_url(url)` function to `socorro/lib/database_utils.py` or something?